src/app/ReadHandler.cpp

```diff
@@ -642,16 +607,10 @@ void ReadHandler::MoveToState(const HandlerState aTargetState)
     // If we just unblocked sending reports, let's go ahead and schedule the reporting
     // engine to run to kick that off.
     //
-    if (aTargetState == HandlerState::GeneratingReports && IsReportableNow())
+    if (aTargetState == HandlerState::GeneratingReports)
```

Comment:
Issue opened here: https://github.com/project-chip/connectedhomeip/issues/28128
OnBecameReportable's purpose has evolved through time since reportability checks are not done in the ReadHandler anymore.
It should be something like: OnReportabilityUpdate or OnReportabilityUnblocked